### PR TITLE
chezmoi: embed version number, include fish completion, improve tests

### PR DESCRIPTION
**Pull request by twpayne**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This builds on @stephengroat's work in #53431 to:

1. Embed the version number, build time, and builder in the binary.
2. Remove an unnecessary build step.
3. Install fish shell completion.
4. Add a test for the version number and builder.

The release bump is because embedding the version number is critical for user support (I am the author of chezmoi).

cc @bayandin who reviewed #53431.
Refs https://github.com/twpayne/chezmoi/pull/697.

Thanks both @stephengroat and @bayandin for getting chezmoi into homebrew-core :)

### CODE REVIEW

File: Formula/chezmoi.rb [hunk 1]
```diff
@@ -15,16 +16,22 @@ class Chezmoi < Formula
   depends_on "go" => :build
 
   def install
-    system "go", "build", *std_go_args, "-ldflags", "-s -w"
+    date = Utils.popen_read("date -u +'%Y-%m-%dT%H:%M:%SZ' 2> /dev/null").chomp
+    system "go", "build", *std_go_args, "-ldflags",
+      "-s -w -X main.version=#{version} -X main.date=#{date} -X main.builtBy=homebrew"
```
Comment (bayandin) [inline Formula/chezmoi.rb:21]:
```suggestion
    commit = Utils.popen_read("git", "rev-parse", "HEAD").chomp
    ldflags = %W[
      -s -w
      -X main.version=#{version}
      -X main.commit=#{commit}
      -X main.date=#{Time.now.utc.rfc3339}
      -X main.builtBy=homebrew
    ].join(" ")
    system "go", "build", *std_go_args, "-ldflags", ldflags
```

File: Formula/chezmoi.rb [hunk 2]
```diff
@@ -3,6 +3,7 @@ class Chezmoi < Formula
   homepage "https://chezmoi.io/"
   url "https://github.com/twpayne/chezmoi/archive/v1.8.0.tar.gz"
   sha256 "b55289372b0e419d9e759f3193cca366a0ae1de2bd72f523d49a0e8ca8567a47"
+  revision 1
   head "https://github.com/twpayne/chezmoi.git"
```
Comment (bayandin) [inline Formula/chezmoi.rb:7]: If we switch to getting repo as a git repo, then we even could set `-X main.commit`

```suggestion
  url "https://github.com/twpayne/chezmoi.git",
      :tag      => "v1.8.0",
      :revision => "017a83f4055a98ac90b030d5739aa560dde239b7"
  revision 1
```

**Review by bayandin — CHANGES_REQUESTED** [review]
Thanks @twpayne!

I suggest a couple of style improvements

**Comment by twpayne** [comment]
Thanks! That's a great improvement. I've merged the changes and will wait for CI to run.

**Comment by bayandin** [comment]
```
==> FAILED
Error: 1 problem in 1 formula detected
chezmoi:
  * stable: sha256 changed without the version also changing; please create an issue upstream to rule out malicious circumstances and to find out why the file changed.
```

This is expected.

@twpayne could you please squash commits into a single one with a message `chezmoi: embed version number, include fish completion, improve tests`. And we're good to go

**Comment by twpayne** [comment]
> @twpayne could you please squash commits into a single one with a message `chezmoi: embed version number, include fish completion, improve tests`.

Done.

**Comment by bayandin** [comment]
Thanks again @twpayne!

**Pull request closed without merging**